mcp/setup.sh

```diff
@@ -111,62 +117,152 @@ installDarwinDependencies(){
+        httpBackend
+        ;;
+      kubernetes )
+        kubernetesBackend
+        ;;
+      local) ;;
+      *);;
+    esac
+    break
+  done
+  echo "}" >> main.tf
+  {
+    echo "module mcp{"
+    echo "  source = \"github.com/mesoform/terraform-infrastructure-modules//mcp\""
+  } >> main.tf
+  setVars
+  echo "}" >> main.tf
+
+  cat main.tf
   echo ""
-  terraform plan -var-file="./settings.tfvars" -out="plan.tfplan"
-  if [ -e plan.tfplan ] 
-  then 
-    read -r -p "Do you wish to apply this plan? (Only 'yes' accepted): " accept
-    case $accept in 
-      yes ) 
+  read -r -p "Are you happy with this configuration? [y]es or [n]o: " accept
+    case $accept in
+      y )
         echo ""
-        echo "Applying Plan..."
+        echo "Setup Complete"
         echo ""
-        terraform apply plan.tfplan 
-        rm plan.tfplan
+        entry
         ;;
-      *)
+      n)
         echo ""
-        echo "Apply Aborted"
+        echo "Configuration rejected"
         echo ""
+        read -r -p "Restart setup? [y]es or [n]o: " restart
+          case $restart in
+          y )
+            echo ""
+            echo "Restarting setup ..."
+            echo ""
+            runSetup
+            ;;
+          *)
+            echo ""
+            echo "Deleting main.tf..."
+            echo ""
+            rm main.tf
+            entry
+            ;;
+          esac
         ;;
+      *)
     esac
-  else
-    echo "Terraform plan failed" | exit 1
-  fi
+  entry
+
+}
+
+gcsBackend(){
+  echo "  backend \"gcs\"{" >> main.tf
+  read -r -p "Enter bucket name: " bucket
+  echo "    bucket = \"${bucket}\"" >> main.tf
+  read -r -p "Enter folder: " folder
+  echo "    folder = \"${folder}\""  >> main.tf
+  echo "  }" >> main.tf
+}
+
+s3Backend(){
+  echo "  backend \"gcs\"{" >> main.tf
+  read -r -p "Enter bucket name: " bucket
+  echo "    bucket = \"${bucket}\"" >> main.tf
+  read -r -p "Enter key: " key
+  echo "    key = \"${key}\""  >> main.tf
+  read -r -p "Enter region: " region
+  echo "    folder = \"${region}\""  >> main.tf
+  echo "  }" >> main.tf
+}
+
+httpBackend(){
+  echo "  backend \"gcs\"{" >> main.tf
+  read -r -p "Enter address: " address
+  echo "    address = \"${address}\"" >> main.tf
+  read -r -p "Enter lock address: " lock_address
+  echo "    lock_address = \"${lock_address}\""  >> main.tf
+  read -r -p "Enter unlock address: " unlock_address
+  echo "    unlock_address = \"${unlock_address}\""  >> main.tf
+  echo "  }" >> main.tf
+}
+
+kubernetesBackend(){
+  echo "  backend \"gcs\"{" >> main.tf
+  read -r -p "Enter secret suffix: " secret_suffix
+  echo "    secret_suffix = \"${secret_suffix}\"" >> main.tf
+  read -r -p "Load config file? (true or false): " config
+  echo "    load_config_file = ${config}"  >> main.tf
+  echo "  }" >> main.tf
+}
+
+setVars(){
+  read -r -p "Enter path to directory of configuration files: " path
+  {
+    echo "  user_project_config_yml = \"${path}/user_project_config.yml\""
```

Comment:
Why do we need all these set?

Reply:
This was to make sure that the right path was set for the yml files, in case the setup script was run in a directory that would make the default path not point to the yml files